csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -825,6 +999,54 @@ private void TestReusingDisposedRunOutput()
             }
         }
 
+        [Fact]
+        private void TestReusingFixedBufferOnnxValueNonStringType()
+        {
+            // model takes 1x5 input of fixed type, echoes back
+            string modelPath = Path.Combine(Directory.GetCurrentDirectory(), "test_types_BOOL.pb");
+            using (var session = new InferenceSession(modelPath))
+            {
+                var bufferA = new bool[] { true, false, true, false, true };
+                var bufferB = new bool[bufferA.Length];
+                var bufferC = new bool[bufferA.Length];
+                var tensorA = new DenseTensor<bool>(bufferA, new int[] { 1, 5 });
+                var tensorB = new DenseTensor<bool>(bufferB, new int[] { 1, 5 });
+                var tensorC = new DenseTensor<bool>(bufferC, new int[] { 1, 5 });
+                using (FixedBufferOnnxValue a = FixedBufferOnnxValue.CreateFromTensor(tensorA),
+                                            b = FixedBufferOnnxValue.CreateFromTensor(tensorB),
+                                            c = FixedBufferOnnxValue.CreateFromTensor(tensorC))
+                {
+                    session.Run(new[] { "input" }, new[] { a }, new[] { "output" }, new[] { b });
+                    session.Run(new[] { "input" }, new[] { b }, new[] { "output" }, new[] { c });
+                }
```

Comment:
Question: Is this the way for users to get maximum speedup? If not, can we update to make it as fast as possible?

Also, why is  the input 'a' only used once -- how to reuse the input? If user sees copies this pattern, and if they want to  score 1000 examples, they  will create 1000 FixedBufferOnnxValues ('a0', 'a1', ... 'a999') . 

Can we add example how the input 'a' can be reused? -- in the example above, it would be same speed as NamedOnnxValue, since we're creating 1000 new objects.

Reply:
I think I need to add a new case

We have 2 different "reuse":
 - use previous output as the next inference's input
 - use one input multiple times 

this test case is for the first one. i will add new test case for second

Reply:
added in TestReusingFixedBufferOnnxValueNonStringTypeMultiInferences()